go/vt/vttablet/tabletserver/schema/engine.go

```diff
@@ -667,6 +667,32 @@ func (se *Engine) GetSchema() map[string]*Table {
 	return tables
 }
 
+// MarshalMinimalSchema returns a protobuf encoded binlogdata.MinimalSchema
+func (se *Engine) MarshalMinimalSchema() ([]byte, error) {
+	se.mu.Lock()
+	defer se.mu.Unlock()
+	dbSchema := &binlogdatapb.MinimalSchema{
+		Tables: []*binlogdatapb.MinimalTable{},
+	}
+	for _, table := range se.tables {
+		dbSchema.Tables = append(dbSchema.Tables, newMinimalTable(table))
+	}
+	return dbSchema.MarshalVT()
+}
+
+func newMinimalTable(st *Table) *binlogdatapb.MinimalTable {
+	table := &binlogdatapb.MinimalTable{
+		Name:   st.Name.String(),
+		Fields: st.Fields,
+	}
+	var pkc []int64
+	for _, pk := range st.PKColumns {
+		pkc = append(pkc, int64(pk))
+	}
```

Comment:
We should be more efficient here and do a single allocation since we know the size (I know this isn't new, but since we're here...):
```suggestion
	pkc := make([]int64, len(st.PKColumns))
	for i, pk := range st.PKColumns {
		pkc[i] = int64(pk)
	}
```

Reply:
done in https://github.com/vitessio/vitess/pull/13045/commits/ee3c5f74f1e95335868b0294f21fd9af59c90689